u2f-gae-demo/src/com/google/u2f/gaedemo/endpoints/U2FRequestHandler.java

```diff
@@ -0,0 +1,246 @@
+    try {
+      u2fSignRequest = u2fServer.getSignRequest(user.getEmail(), Constants.APP_ID);
+    } catch (U2FException e) {
+      throw new ServletException("couldn't get sign request", e);
+    }
+
+    JsonObject result = new JsonObject();
+    result.addProperty("challenge", u2fSignRequest.getChallenge());
+    result.addProperty("appId", Constants.APP_ID);
+    result.add("registeredKeys", u2fSignRequest.getRegisteredKeysAsJson(Constants.APP_ID));
+
+    return new String[] {result.toString()};
+  }
+
+  @ApiMethod(name = "processSignResponse")
+  public String[] processSignResponse(@Named("responseData") String responseData, User user)
+      throws OAuthRequestException, ServletException {
+    if (user == null) {
+      throw new OAuthRequestException("user is not authenticated");
+    }
+
+    JsonObject responseDataJson = (JsonObject) new JsonParser().parse(responseData);
+    String currentUser = user.getEmail();
+    String expectedUser =
+        dataStore.getSignSessionData(responseDataJson.get("sessionId").toString()).getAccountName();
+
+    if (!currentUser.equals(expectedUser)) {
+      throw new ServletException("Cross-site request prohibited");
+    }
+
+    SignResponse signResponse =
+        new SignResponse(
+            responseDataJson.get("keyHandle").toString(),
+            responseDataJson.get("signatureData").toString(),
+            responseDataJson.get("clientData").toString(),
+            responseDataJson.get("sessionId").toString());
+
+    if (u2fServer == null) {
+      initU2fServer();
+    }
+
+    SecurityKeyData securityKeyData;
+    try {
+      securityKeyData = u2fServer.processSignResponse(signResponse);
+    } catch (U2FException e) {
+      throw new ServletException("signature didn't verify", e);
+    }
+
+    return new String[] {new TokenStorageData(securityKeyData).toJson().toString()};
+  }
+
+  @ApiMethod(name = "getAllSecurityKeys", path = "getAllSecurityKeys")
+  public String[] getAllSecurityKeys(User user) throws OAuthRequestException {
+    if (user == null) {
+      throw new OAuthRequestException("user is not authenticated");
+    }
+
+    JsonArray resultList = new JsonArray();
+    if (u2fServer == null) {
+      initU2fServer();
+    }
+
+    List<SecurityKeyData> tokens = u2fServer.getAllSecurityKeys(user.getEmail());
+    for (SecurityKeyData token : tokens) {
+      resultList.add(new TokenStorageData(token).toJson());
+    }
+    return new String[] {resultList.toString()};
+  }
+
+  @ApiMethod(name = "removeSecurityKey")
+  public String[] removeSecurityKey(User user, @Named("publicKey") String publicKey)
+      throws OAuthRequestException, U2FException, DecoderException {
+    if (user == null) {
+      throw new OAuthRequestException("user is not authenticated");
+    }
+
+    if (u2fServer == null) {
+      initU2fServer();
+    }
+
+    u2fServer.removeSecurityKey(user.getEmail(), Hex.decodeHex(publicKey.toCharArray()));
+    return new String[] {"OK"};
```

Comment:
Returning a simple type such as String or int is not supported. The return value needs to be a JavaBean (non-parametrized), an array or a Collection. 
https://cloud.google.com/appengine/docs/standard/java/endpoints/parameter-and-return-types

I chose to use simple array for all method return types; and for this method specifically, I thought using "OK" to represent the successful completion of that operation, the caller can choose to examine this status.